crates/fluvio-spu/src/services/public/mod.rs

```diff
@@ -76,100 +78,119 @@ impl FluvioService for PublicService {
+                            ),
+                            SpuServerRequest::UpdateConsumerOffsetRequest(request) => {
+                                call_service!(
+                                    request,
+                                    handle_update_consumer_offset_request(
+                                        request,
+                                        context.clone(),
+                                        &mut conn_ctx
+                                    ),
+                                    shared_sink,
+                                    "UpdateConsumerRequest"
+                                )
+                            }
+                            SpuServerRequest::DeleteConsumerOffsetRequest(request) => {
+                                call_service!(
+                                    request,
+                                    handle_delete_consumer_offset_request(request, context.clone()),
+                                    shared_sink,
+                                    "DeleteConsumerRequest"
+                                )
+                            }
+                            SpuServerRequest::FetchConsumerOffsetsRequest(request) => {
+                                call_service!(
+                                    request,
+                                    handle_fetch_consumer_offsets_request(request, context.clone()),
+                                    shared_sink,
+                                    "FetchConsumersRequest"
+                                )
+                            }
+                            SpuServerRequest::StartMirrorRequest(request) => {
+                                // send mirror mode, afer that mirror cycle will be started
+                                mirror_request = Some(request);
+                                break;
+                            }
+                        }
+                    }
+                    Some(Err(e)) => {
+                        debug!(
+                            sink_id = shared_sink.id(),
+                            "Error decoding message, ending connection: {}", e
+                        );
+                        break;
+                    }
+                    None => {
+                        debug!(sink_id = shared_sink.id(), "No content, end of connection",);
+                        break;
                     }
-                }
-                Some(Err(e)) => {
-                    debug!(
-                        sink_id = shared_sink.id(),
-                        "Error decoding message, ending connection: {}", e
-                    );
-                    break;
-                }
-                None => {
-                    debug!(sink_id = shared_sink.id(), "No content, end of connection",);
-                    break;
                 }
             }
         }
 
+        if let Some(request) = mirror_request {
```

Comment:
Why can't we respond in place? Why saving the request to var and process outside the loop?